database/models/progress.ts

```diff
@@ -0,0 +1,33 @@
+import { model, models, Schema, Types } from "mongoose";
+
+export interface ProgressInterface {
+    _id: Types.ObjectId;
+    username: string;
+    module: string;
+    completion: number;
+    input: Schema.Types.Mixed;
```

Comment:
I think the should just be `any` as a type.

Reply:
Additionally, it's not `required` in the schema so it should be optional as a type

Reply:
Using type 'any' doesn't work, so I'm keeping mixed for now (mixed seems to be the standard for mongoose schemas).